class.jetpack-gutenberg.php

```diff
@@ -10,143 +10,152 @@
+ * @param string $reason A string representation of why the extension is unavailable
+ *
+ * @since 7.0.0
+ *
+ * @return void
+ */
+function jetpack_set_extension_unavailability_reason( $slug, $reason ) {
+	Jetpack_Gutenberg::set_extension_unavailability_reason( $slug, $reason );
 }
 
 /**
  * General Gutenberg editor specific functionality
  */
 class Jetpack_Gutenberg {
 
-	// BLOCKS
-	private static $default_blocks = array(
-		'map',
-		'markdown',
-		'simple-payments',
-		'related-posts',
-		'contact-form',
-		'field-text',
-		'field-name',
-		'field-email',
-		'field-url',
-		'field-date',
-		'field-telephone',
-		'field-textarea',
-		'field-checkbox',
-		'field-checkbox-multiple',
-		'field-radio',
-		'field-select',
-		'subscriptions',
-	);
-
 	/**
-	 * @var array Array of blocks information.
+	 * @var array Extensions whitelist
 	 *
-	 * For each block we have information about the availability for the current user
+	 * Only these extensions can be registered. Used to control availability of beta blocks.
 	 */
-	private static $blocks = array();
+	private static $extensions = array();
 
+	/**
+	 * @var array Extensions availability information
+	 *
+	 * Keeps track of the reasons why a given extension is unavailable.
+	 */
 	private static $availability = array();
 
-	// PLUGINS
-	private static $default_plugins = array(
-		'publicize',
-		'shortlinks',
-	);
 	/**
-	 * @var array Array of plugins information.
+	 * @var array Plugin registry
 	 *
-	 * For each block we have information about the availability for the current user
+	 * Since there is no `register_plugin()` counterpart to `register_block_type()` in Gutenberg,
+	 * we have to keep track of plugin registration ourselves
 	 */
-	private static $plugins = array();
+	private static $registered_plugins = array();
 
 	/**
-	 * @var array Array of extensions we will be registering.
+	 * Prepend the 'jetpack/' prefix to a block name
+	 *
+	 * @param string $block_name The block name
+	 *
+	 * @return string The prefixed block name.
 	 */
-	private static $registered = array();
+	private static function prepend_block_prefix( $block_name ) {
+		return 'jetpack/' . $block_name;
+	}
 
 	/**
-	 * Add a block to the list of blocks to be registered.
+	 * Whether two arrays share at least one item
 	 *
-	 * @param string $slug Slug of the block.
-	 * @param array  $args Arguments that are passed into the register_block_type.
-	 * @param array $availability array containing if a block is available and the reason when it is not.
+	 * @param array $a An array
+	 * @param array $b Another array
+	 *
+	 * @return boolean True if $a and $b share at least one item
 	 */
-	public static function register( $slug, $args, $availability ) {
-		$sanitized_slug = sanitize_title_with_dashes( $slug );
-		self::$registered[ $sanitized_slug ] = array( 'args' => $args, 'availability' => $availability );
+	protected static function share_items( $a, $b ) {
+		return count( array_intersect( $a, $b ) ) > 0;
```

Comment:
Equivalent and more descriptive?

```suggestion
		return ! empty( array_intersect( $a, $b ) );
```

Reply:
I had it like that, but then I was reminded that prior to PHP 5.5, `empty()` would only work for variables (but not for expressions) and would apparently yield a parse error for anything else.

[SO thread](https://stackoverflow.com/questions/523796/checking-to-see-if-one-arrays-elements-are-in-another-array-in-php) about checking if intersections of arrays are empty where this is coincidentally brought up.